### PR TITLE
Beacon interactions with gallery

### PR DESCRIPTION
**Pull request by ironsidevsquincy**
Beacon when the gallery loads, and when the user interacts

### CODE REVIEW

File: client/components/slideshow/main.js [hunk 1]
```diff
@@ -10,14 +11,27 @@ module.exports = function(els) {
...
 					container.innerHTML = data;
 					el.parentNode.replaceChild(container, el);
 					return container;
 				})
 				.then(function(el) {
+					var totalItems;
 					el.style.width = el.clientWidth + 'px';
+					el.addEventListener('oGallery.itemSelect', function (ev) {
+						if (ev.target.classList.contains('o-gallery--slideshow')) {
+							Analytics.fire('galleries', { item: ev.detail.itemID, totalItems: totalItems, interacted: true });
```
Comment (commuterjoy) [inline client/components/slideshow/main.js]: Can this be `gallery` please?

Comment (commuterjoy) [inline client/components/slideshow/main.js]: instead of item i'd be more specific - i.e. `picture`

**Comment by matthew-andrews** [comment]
oh sorry……

File: client/components/slideshow/main.js [hunk 2]
```diff
@@ -10,14 +11,27 @@ module.exports = function(els) {
...
+						}
+					});
+					el.addEventListener('oGallery.ready', function oGalleryIsReady(ev) {
+						if (ev.target.classList.contains('o-gallery--slideshow')) {
+							el.removeEventListener(ev.type, oGalleryIsReady);
+							totalItems = ev.target.querySelectorAll('.o-gallery__item').length;
+							Analytics.fire('galleries', { item: 0, totalItems: totalItems, interacted: false });
```
Comment (commuterjoy) [inline client/components/slideshow/main.js]: what's the idea with this?

Comment (ironsidevsquincy) [inline client/components/slideshow/main.js]: a way of knowing which pages have a gallery on them. maybe better to add it to the `page` data?

File: client/components/slideshow/main.js (hunk 1 above)
Comment (ironsidevsquincy) [inline client/components/slideshow/main.js]: I'll singularise™ and change to a more human word

File: client/components/slideshow/main.js (hunk 2 above)
Comment (commuterjoy) [inline client/components/slideshow/main.js]: Do this on the `next-beacon` project, on the server. I want to move away from using the client to track *everything* when it's more obvious to do on the server.

Each beacon call hits CAPI, so you could just add a `hasGallery` flag to the page.capi object

https://github.com/Financial-Times/next-beacon/blob/master/models/event.js#L84

**Pull request merged**